web/src/main/resources/assets/js/graphhopper/GHRequest.js

```diff
@@ -118,34 +119,60 @@ GHRequest.prototype.getEarliestDepartureTime = function () {
     return undefined;
 };
 
-GHRequest.prototype.initVehicle = function (vehicle) {
-    this.api_params.vehicle = vehicle;
-    if(this.api_params.elevation !== false) {
-        var featureSet = this.features[vehicle];
-        this.api_params.elevation = featureSet && featureSet.elevation;
-    }
-    this.hasTCSupport();
+GHRequest.prototype.setProfile = function (profileName) {
+    this.api_params.profile = profileName;
+};
+
+GHRequest.prototype.getProfile = function () {
+    return this.api_params.profile;
 };
 
-GHRequest.prototype.hasTCSupport = function() {
-   if(this.api_params.turn_costs !== false) {
-      var featureSet = this.features[this.api_params.vehicle];
-      this.api_params.turn_costs = featureSet && featureSet.turn_costs;
-   }
+GHRequest.prototype.setElevation = function (elevation) {
+    this.api_params.elevation = elevation;
 };
 
 GHRequest.prototype.hasElevation = function () {
     return this.api_params.elevation === true;
 };
 
 GHRequest.prototype.getVehicle = function () {
-    return this.api_params.vehicle;
+    var profileName = this.api_params.profile;
+    var profile = this.profiles.find(function(p) { return p.profile_name === profileName; });
+    if (!profile)
+        return "";
+    else
+        return profile.vehicle;
 };
 
 GHRequest.prototype.isPublicTransit = function () {
-    return this.getVehicle() === "pt";
+    // legacy support: we might have set vehicle=pt instead of pt
+    return this.getProfile() === "pt" || this.getVehicle() === "pt";;
 };
 
+GHRequest.prototype.removeProfileParameterIfLegacyRequest = function() {
+     // we still allow using legacy parameters to support older urls pasted from somewhere, but when they are used
+     // we may not add the profile parameter to the url
+     if (
+            this.api_params["vehicle"] ||
+            this.api_params["weighting"] ||
+            this.api_params["turn_costs"] ||
+            this.api_params["edge_based"]
+        ) {
+            delete this.api_params.profile;
+        }
+     // todonow: pasting a legacy url already works the only thing that is ugly is that when we do a normal request
+     // (including profile) then add something like vehicle=car the the profile parameter appears still in the url field
+     // (even though it will not be sent to the server). when we drag a marker afterwards it finally disappears, how can
+     // we do this already here?
```

Comment:
@karussell would you mind taking a look at this? I did not manage to fix this and not even sure if this needs to be fixed.

Reply:
the problem is this:

1) use the UI (profile will be in the address line)
2) manually add something like `vehicle=car` and press enter
3) the profile will not be send to the server (only the vehicle), but the address line is not updated accordingly
4) move a marker. same as 3) but this time the profile is removed from the address line

> And shouldn't we remove the vehicle parameter from the URL if vehicle and profile parameter exist in the URL?

Yes that sounds right. So we only support 'real' legacy requests (without profile) and the (now) standard (including (only) a profile). I probably tried this but ran into similar problems as above?

Reply:
> manually add something like vehicle=car and press enter

Ah, ok. Then this is not a legacy use case and either the profile parameter needs to be removed or instead of the vehicle the profile needs to be changed.

> So we only support 'real' legacy requests (without profile) and the (now) standard (including (only) a profile)

Yes

> I probably tried this but ran into similar problems as above?

Unfortunately I was not able to get this mixture (yet) using a legacy URL.

Reply:
> Ah, ok. Then this is not a legacy use case and either the profile parameter needs to be removed or instead of the vehicle the profile needs to be changed.

Yeah its really a very minor issue / very exotic '(mis)-use-case'. I was just annoyed by the fact that moving the marker does something that pressing enter does not do. Even when doing this (add vehicle to a profile url manually) the UI keeps working (using the vehicle), it just does not remove the profile from the url (and yes maybe it would better remove the vehicle in this case but this is probably the same problem).

Reply:
Shall we do anything about this?

Reply:
I tried to deploy this code to production and had a problem that although only profile=car was specified in the URL still the weighting was added in the request (resulting in an error). It could be that this is some internal problem as the /info endpoint is not yet updated. Will need to investigate this further but we can also merge here and fix later.